### ISSUE TITLE
2.1.1.0 Early Game Bugs and Nitpicks and Other Things

### ISSUE THREAD

**Issue by ScriptedPiky** [GTNewHorizons/GT-New-Horizons-Modpack#8238]
All of these thoughts & such have occurred while playing with a friend on a private server. We decided to boot up a fresh instance of GTNH and played through the very beginning and now approach basic LV.

**(Tinkers and GregTech or Dreamcraft/Coremod or Scripts?)** Certain Green and Blue Slime (doesn't affect pink crystalline slime) Tinker's parts cannot be crafted normally or have extremely weird recipe interactions. Every single one of the Blue Slime extruder recipes has this issue of being unable, additionally, other parts cannot be crafted (maybe intended?).
![2021-07-13_19 34 03](https://user-images.githubusercontent.com/41557212/125571058-173dd291-c116-48b3-bd68-e023c22fda9d.png)

**(SpecialMobs Mod)** 1# The player can die twice to a Brutish Mob that has either the Pain or Agony enchantment.
2# Witch of the Wind should either have a 10-second teleportation cooldown or have its teleportation removed entirely.
3# Jolt Blazes should have a cooldown on their lightning bolt summon instead of summoning a lightning bolt for every tick of damage(!) they take.
4# Witch of the Undead should also have a slightly longer cooldown on how often they can summon mobs
5# Skeletons with the arrow that spawns a torch if it hits a surface block(?) do not deal damage to entities/the player, effectively harmless until they "run out of ammo;" this part is still extremely uncertain as to what happens or is causing this.
6# Unholy Ghasts have unbelievably far attack range for a melee attack despite their small size, often quoted as "extended lollipop."

**(Quests)**
"Time to kill Slime" and "Time to kill Magma Slime" and "Time to kill Dire Wolf" quests should have the `The "Mechanic" Coin` replaced with `The "Farmer" Coin`. This should allow for certain farming coin quests to be more forgiving in the early game, such as the Woven Cotton coin quest (it gives string and wool) or sapling bags for the many rare saplings throughout Pam's Harvest Craft, as those are not easily obtainable without. Additionally, there could be a quest rework for exhausting large amounts of basic resources for a reword of one of many Pam's Harvest Craft saplings (if there isn't a sapling bag yet, which I might've contradicted myself); the "Olive Tree" coin quest could be altered into this such instance of offering the player a reward into being able to pick one of the many saplings.

**Comment by Methes** [comment]
I vote for removing teleportation from both Witch of the Wind and Jolt Blazes - it fucks up any mob farm having those mobs and there's nothing one can do about it.

**Comment by GTNH-Colen** [comment]
Wow it's almost like infernal mobs is a laggy, unbalanced, buggy mod held together by duck tape. Who could've guessed. 

"Dying 15 times to a mob that deletes your grave is so fun guys, why change anything?" - User in infinity armour

**Comment by boubou19** [comment]
i thought mobs deleting grave was fixed? or was it hungry nodes? can't remember

**Comment by ScriptedPiky** [comment]
Maybe both were fixed at one point? I know that Moron fixed SpecialMobs and the obscene code it had, though it might still be possible in a very rogue instance. Although, it current can kill the player twice and spawn a grave twice, however, the first grave will contain the player's items and the other will contain just a head

**Comment by boubou19** [comment]
it's linked to the enchant i'm sure, as it also happens when players kill players with those wooden swords.

**Comment by ScriptedPiky** [comment]
Definitely linked to the enchantment it seems, as it tries to apply another instance of damage(?) the very next tick upon a strike successfully being dealt

**Comment by Alastors** [comment]
Im down for infernal mods to have a lot of it's more annoying mob modifiers to be removed tbh. For example, vengeance deals the same amount of damage to you that it did the mob, not the damage based on the mob's health. So if you one-shotted a creeper with 10 hearts for 70 hearts worth of crossbow, it takes the 10 that kills it, but you take the full 70 hearts worth of damage. (without armour, though this may still kill those in extremely good armour.) Which is very dumb and broken and 100% should be fixed or turned off tbh.

**Comment by D-Cysteine** [comment]
Yeah, the vengeance modifier also was an outlier for me. As soon as I got a decent crossbow (vanadiumsteel), most mobs weren't a threat, but a single shot on vengeance mobs was causing me to be left with a handful of hearts and having to wait ages to heal, or even one-shotting myself. This was with something like 30 hearts and full iron armor.

To make things worse, mobs at the far edge of visible distance are too far away to show modifiers, so for example shooting far-away ghasts in the Nether was just a gamble on whether or not it happened to be vengeance. Another reason I don't like the vengeance mechanic is that it kind of feels like a punishment for getting better weapons: with a weaker weapon, say one that just barely kills the mob, the total reflected damage will be about equal to the mob's max HP, whereas with a really strong weapon that overkills the mob, the reflected damage could be much greater than the mob's max HP.

As suggested in the preceding comment, instead of just outright disabling vengeance, a less drastic option would be to limit the reflected damage to the mob's current HP. I think this would be pretty simple to do; we just need to take a `Math.min()` with `mob.getHealth()` here:
https://github.com/GTNewHorizons/Infernal-Mobs/blob/1.7.10/src/main/java/atomicstryker/infernalmobs/common/mods/MM_Vengeance.java#L27-L35
This option also fixes the "punishment" issue since stronger weapons will no longer result in more damage reflected than weaker weapons.
